### PR TITLE
Add search to formula

### PR DESCRIPTION
**Pull request by flash1293**
Adds a search input to the formula help

### CODE REVIEW

**Comment by MichaelMarcialis** [comment]
Reviewing now. Will provide a design PR with some of my suggestions shortly.

**Comment by MichaelMarcialis** [comment]
Hey, @flash1293. Per my previous comment, I've made a quick design PR for this branch to tweak some of the styles. Please review here when you get a moment: https://github.com/flash1293/kibana/pull/13

Additionally, a few questions came up during my review of this PR:

1. I noticed that the headings for the docs' nav groups are always showing, regardless the currently entered search string returns no matching child functions for that group and that string is not contained within the heading. Should these headings be shown given those circumstances? My initial instinct is that we shouldn't be showing them in that case.

image

2. While individual function items are given an `EuiMark` in order to show the portion of the search string contained in each item, no such indicator is provided to nav group headings. Should the matching search strings within the nav group headings be stylistically offset to show the matching search string like we do for function items?

image

**Comment by flash1293** [comment]
@MichaelMarcialis Made the section headers searchable as well (including hiding them if nothing in a section matches current search)

Screenshot 2021-05-26 at 14 12 41

File: x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/formula/editor/formula_help.tsx [hunk 1]
```diff
@@ -15,10 +15,12 @@ import {
   EuiText,
   EuiListGroupItem,
   EuiListGroup,
-  EuiMarkdownFormat,
   EuiTitle,
   EuiButtonIcon,
+  EuiFieldSearch,
+  EuiHighlight,
 } from '@elastic/eui';
+import { Markdown } from '../../../../../../../../../src/plugins/kibana_react/public';
```
Comment (wylieconlon) [inline x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/formula/editor/formula_help.tsx:23]: I thought we agreed on EuiMarkdownFormat @flash1293?

**Review by wylieconlon — APPROVED** [review]
@flash1293 LGTM, just curious about the markdown switch.

**Comment by MichaelMarcialis** [comment]
> @flash1293 LGTM, just curious about the markdown switch.

The `EuiMarkdownFormat` component doesn't appear to offer any control over the text styling, and wrapping it inside an `EuiText` component was yielding inconsistent results in some instances. It was more controllable and easier to maintain by using the `Markdown` component instead and wrapping the entire output in a single `EuiText`.

**Pull request merged**